src/mds/Server.cc

```diff
@@ -1118,6 +1118,15 @@ void Server::find_idle_sessions()
 	     << "), not marking any client stale" << dendl;
     return;
   }
+  // don't kick client if ops or osds are laggy
+  bool laggy = mds->objecter->with_osdmap([](const OSDMap &map) {
+    return map.any_osd_laggy(); });
+  if(laggy) {
+    mds->get_beacon().notify_health(mds);
+    dout(10) << "laggy OSDs deteced, skipping laggy clients eviction " 
+             << "untill they return to normal" << dendl;
+    return;
+  }
```

Comment:
Only do this is if there are any sessions in `to_evict`. (This is the "dry-run" we talked about.)

Reply:
Right, but the only concern was the extra time complexity associated with iterating on all the sessions(https://github.com/ceph/ceph/pull/49971#discussion_r1126181281), this information seems like non-trivial, I'll make this change

Reply:
The MDS would iterate the sessions if the OSDs were not laggy anyway. This is not changing the time complexity of the method.

Reply:
had a discussion on this with 1x1 with @gregsfortytwo last week and I think this is a crucial piece of information, would definitely come in handy for users/admins to diagnose issues with clients. Therefore, I've pushed the relevant code.